### ISSUE TITLE
[BUG][Ascend] Lost stores: masked store skips 2xfp32 vector groups nondeterministically

### ISSUE THREAD

**Issue by O1iveira7** [flagos-ai/FlagTree#1245]
## Environment
- Triton 3.5.1 (FlagTree 0.7.0rc2+ascend3.5), Ascend (NPU) backend
- torch 2.9.0+cpu + torch_npu 2.9.0.post2, CANN 9.0.0 (driver 25.2.0), Python 3.11.15
- Ascend 910B
## Symptom
在 Ascend 910B4（CANN 9.0.0）+ FlagTree ascend3.5（Triton 3.5.1）上，含 masked tl.store 的 Triton kernel 会出现静默丢写：kernel 编译、执行均无任何报错，但 torch.npu.synchronize() 返回后，部分输出元素仍保持运行前的值（将输出预填充为 NaN 即可观察到残留），且该现象具有非确定性——同一输入重复执行时，丢失的位置和数量逐次变化，部分配置可稳定复现（20/20），部分仅偶发（约 3/20）；丢失以连续 2×fp32（8 字节）的向量组为单位，频率随 store 有效宽度（向量组数量）单调上升（同一 kernel 在 cols=1/2/3/4/8 下 20 次执行的失败次数为 0/2/0/10/17）；再次同步并执行其他 kernel 后重新读取，残留值不变，说明是 store 未执行或未提交，而非读侧可见性问题；原始 kernel 与语义保持的变体均可出现该现象，且全程无崩溃、无异常、无错误码，只能依靠输出比对或哨兵值发现。
**FlagTree Ascned3.5 output:**
```
run 0: unwritten=8
run 1: unwritten=4
run 2: unwritten=4
run 3: unwritten=6
run 4: unwritten=2
run 5: unwritten=4
run 6: unwritten=2
run 7: unwritten=2
run 8: unwritten=12
run 9: unwritten=6
run 10: unwritten=4
run 11: unwritten=2
run 12: unwritten=8
run 13: unwritten=4
run 14: unwritten=2
run 15: unwritten=4
run 16: unwritten=6
run 17: unwritten=8
run 18: unwritten=6
run 19: unwritten=8
summary: bad_runs=20/20
```

**Triton3.7 RTX3070 output:**
```
summary: bad_runs=0/20
[not reproduced] all 20 runs wrote every output element
EXIT=1
```

  ## Minimal reproducer
```
import torch
import triton
import triton.language as tl


@triton.jit
def matrix_transpose_kernel(
    input,
    output,
    rows,
    cols,
    stride_ir,
    stride_ic,
    stride_or,
    stride_oc,
    BLOCK_N: tl.constexpr,
    BLOCK_M: tl.constexpr,
):
    pid_m = tl.program_id(0)
    pid_n = tl.program_id(1)

    row_offsets = pid_m * BLOCK_M + tl.arange(0, BLOCK_M)
    col_offsets = pid_n * BLOCK_N + tl.arange(0, BLOCK_N)

    x_ptrs = input + row_offsets[:, None] * stride_ir + col_offsets[None, :]
    mask = (row_offsets[:, None] < rows) & (col_offsets[None, :] < cols)

    tile = tl.load(x_ptrs, mask=mask, other=0.0)

    y_ptrs = output + col_offsets[:, None] * stride_or + row_offsets[None, :]
    tl.store(y_ptrs, tile.T, mask=mask.T)


def solve(input: torch.Tensor, output: torch.Tensor, rows: int, cols: int) -> None:
    stride_ir, stride_ic = (cols, 1)
    stride_or, stride_oc = (rows, 1)
    grid = (rows, cols)
    matrix_transpose_kernel[grid](
        input,
        output,
        rows,
        cols,
        stride_ir,
        stride_ic,
        stride_or,
        stride_oc,
        32,
        32,
    )
    torch.npu.synchronize()


def main() -> int:
    assert torch.npu.is_available()
    rows, cols = 2, 8
    x = torch.arange(rows * cols, dtype=torch.float32, device="npu").reshape(rows, cols)
    bad_runs = 0
    for index in range(20):
        out = torch.full((cols, rows), float("nan"), dtype=torch.float32, device="npu")
        solve(x, out, rows, cols)
        unwritten = int(torch.isnan(out).sum().item())
        if unwritten:
            bad_runs += 1
            print(f"run {index}: unwritten={unwritten}")
    print(f"summary: bad_runs={bad_runs}/20")
    if bad_runs:
        print("[reproduced] dropped masked stores (output elements left unwritten)")
        return 0
    print("[not reproduced] all 20 runs wrote every output element")
    return 1


if __name__ == "__main__":
    raise SystemExit(main())
 ```

**Comment by O1iveira7** [comment]
The following code may have the same issue
```
import torch
import triton
import triton.language as tl

@triton.jit
def _fz_outlined__causal_dw_conv1d_kernel_15(_fz_a4_bc_x):
    return _fz_a4_bc_x.to(tl.float32, bitcast=True)

@triton.jit
def _causal_dw_conv1d_kernel(_fz_switch_ptr, _fz_switch_output_ptr, x_ptr, w_ptr, b_ptr, out_ptr, B, L, D, K, stride_xb, stride_xl, stride_xd, stride_wd, stride_wk, n_l_blocks, BLOCK_D: tl.constexpr):
    opaque_values = tl.load(_fz_switch_ptr + tl.arange(0, 2))
    opaque_zero, opaque_one = tl.split(opaque_values)
    opaque_true = opaque_zero < opaque_one
    opaque_false = opaque_zero > opaque_one
    pid = tl.program_id(0)
    db_id = pid // n_l_blocks
    bl_id = pid % n_l_blocks
    if opaque_false:
        _fz_a20_split_1 = opaque_one - opaque_zero
        _fz_a12_add = opaque_zero + opaque_one
        _fz_a12_sub = _fz_a20_split_1
        _fz_a12_mul = (_fz_a12_add + 1) * (_fz_a12_sub + 1)
        _fz_a12_float = _fz_a12_mul.to(tl.float32)
        if opaque_false:
            _fz_a22_cf_x = tl.arange(0, 8) + opaque_one
            _fz_a22_cf_a = tl.add(_fz_a22_cf_x, _fz_a22_cf_x)
            _fz_a22_cf_s = _fz_a22_cf_a - _fz_a22_cf_x
            _fz_a22_cf_m = _fz_a22_cf_s * 3
            _fz_a22_cf_c = tl.cast(_fz_a22_cf_m, tl.float32)
            _fz_a22_cf_t = tl.full((), 2.0, tl.float32)
            _fz_a22_cf_f = _fz_a22_cf_c * _fz_a22_cf_t
            _fz_a22_cf_sum = tl.sum(_fz_a22_cf_f)
            _fz_a22_tpl_result = _fz_a22_cf_sum + opaque_values.to(tl.float32)
            tl.store(_fz_switch_output_ptr + 42 + tl.arange(0, 2), _fz_a22_tpl_result, mask=(tl.program_id(axis=0) == 0) & (tl.program_id(axis=1) == 0) & (tl.program_id(axis=2) == 0))
        if opaque_false:
            _fz_a12_div = _fz_a12_float / (opaque_one + 1)
        else:
            _fz_a12_div = _fz_a12_float / (opaque_one + 1)
        _fz_a12_row = tl.expand_dims(opaque_values + _fz_a12_div, 0)
        _fz_a12_transposed = tl.trans(_fz_a12_row)
        _fz_a12_sin = tl.sin(_fz_a12_transposed)
        _fz_a12_tpl_result = tl.ravel(tl.cos(_fz_a12_sin))
        tl.store(_fz_switch_output_ptr + 22 + tl.arange(0, 2), _fz_a12_tpl_result, mask=(tl.program_id(axis=0) == 0) & (tl.program_id(axis=1) == 0) & (tl.program_id(axis=2) == 0))
        return
    if opaque_false:
        _fz_a9_smx_x = tl.arange(0, 8).to(tl.float32) + opaque_one.to(tl.float32)
        _fz_a9_smx_y = tl.softmax(_fz_a9_smx_x)
        _fz_a9_smx_z = tl.exp(_fz_a9_smx_y) - tl.log(_fz_a9_smx_y + 1.0)
        if opaque_false:
            _fz_a9_smx_s = tl.sum(_fz_a9_smx_z)
        else:
            _fz_a9_smx_s = tl.sum(_fz_a9_smx_z)
        _fz_a9_tpl_result = _fz_a9_smx_s + opaque_values.to(tl.float32)
        _fz_a10_loop_once = 0
        if opaque_true:
            tl.store(_fz_switch_output_ptr + 16 + tl.arange(0, 2), _fz_a9_tpl_result, mask=(tl.program_id(axis=0) == 0) & (tl.program_id(axis=1) == 0) & (tl.program_id(axis=2) == 0))
        else:
            pass
        return
    l = bl_id % L
    b = bl_id // L
    d_offs = db_id * BLOCK_D + tl.arange(0, BLOCK_D)
    if opaque_true:
        mask = d_offs < D
    else:
        mask = d_offs < D
    acc = tl.load(b_ptr + d_offs, mask=mask, other=0.0)
    for k in range(K):
        w = tl.load(w_ptr + d_offs * stride_wd + k * stride_wk, mask=mask, other=0.0)
        _fz_a3_split_1 = l - k
        _fz_a3_split_1 = l - k
        _fz_a3_split_1 = l - k
        _fz_a24_split_3 = _fz_a3_split_1
        l_in = _fz_a24_split_3
        if opaque_true:
            _fz_a2_mm_x = tl.arange(0, 8).to(tl.float32) - 3.0 + opaque_one.to(tl.float32)
            _fz_a2_mm_y = tl.arange(0, 8).to(tl.float32) % 4.0 + opaque_one.to(tl.float32)
            _fz_a2_mm_a = tl.minimum(_fz_a2_mm_x, _fz_a2_mm_y)
            _fz_a2_mm_b = tl.maximum(_fz_a2_mm_x, _fz_a2_mm_y)
            _fz_a2_mm_s = tl.sum(_fz_a2_mm_a + _fz_a2_mm_b)
            _fz_a2_tpl_result = _fz_a2_mm_s + opaque_values.to(tl.float32)
            _fz_a6_loop_once = 0
            if opaque_true:
                for _fz_a7_loop_once in range(0, 1):
                    if opaque_true:
                        tl.store(_fz_switch_output_ptr + 2 + tl.arange(0, 2), _fz_a2_tpl_result, mask=(tl.program_id(axis=0) == 0) & (tl.program_id(axis=1) == 0) & (tl.program_id(axis=2) == 0))
                    else:
                        pass
            else:
                pass
        if l_in >= 0:
            if opaque_true:
                x_val = tl.load(x_ptr + b * stride_xb + l_in * stride_xl + d_offs * stride_xd, mask=mask, other=0.0)
            else:
                x_val = tl.load(x_ptr + b * stride_xb + l_in * stride_xl + d_offs * stride_xd, mask=mask, other=0.0)
            acc = acc + w * x_val
    _fz_a1_loop_once = 0
    if opaque_true:
        tl.store(out_ptr + b * stride_xb + l * stride_xl + d_offs * stride_xd, acc, mask=mask)
    else:
        if opaque_false:
            if opaque_false:
                _fz_a11_f16_x = tl.arange(0, 8).to(tl.float16) + opaque_one.to(tl.float16)
                _fz_a11_f16_y = _fz_a11_f16_x * 1.5 - 0.5
                _fz_a11_f16_z = _fz_a11_f16_y.to(tl.float32) + _fz_a11_f16_x.to(tl.float32)
                _fz_a11_f16_w = _fz_a11_f16_z.to(tl.bfloat16).to(tl.float32)
                _fz_a11_tpl_result = tl.sum(_fz_a11_f16_w / 4.0) + opaque_values.to(tl.float32)
                tl.store(_fz_switch_output_ptr + 20 + tl.arange(0, 2), _fz_a11_tpl_result, mask=(tl.program_id(axis=0) == 0) & (tl.program_id(axis=1) == 0) & (tl.program_id(axis=2) == 0))
            _fz_a4_bc_x = tl.arange(0, 8) + opaque_one
            _fz_a4_bc_f = _fz_outlined__causal_dw_conv1d_kernel_15(_fz_a4_bc_x)
            _fz_a4_bc_i = _fz_a4_bc_f.to(tl.int32, bitcast=True)
            _fz_a4_bc_h = (_fz_a4_bc_i + 1).to(tl.int16).to(tl.float16, bitcast=True)
            _fz_a4_bc_u = _fz_a4_bc_h.to(tl.uint16, bitcast=True)
            _fz_a4_bc_h2 = _fz_a4_bc_u.to(tl.float16, bitcast=True)
            _fz_a4_bc_b = (_fz_a4_bc_i + 2).to(tl.int16).to(tl.bfloat16, bitcast=True)
            _fz_a4_bc_b2 = _fz_a4_bc_b.to(tl.uint16, bitcast=True).to(tl.bfloat16, bitcast=True)
            _fz_a4_bc_s = tl.sum((_fz_a4_bc_i - _fz_a4_bc_x).to(tl.float32) + _fz_a4_bc_h2.to(tl.float32) + _fz_a4_bc_b2.to(tl.float32))
            _fz_a4_tpl_result = _fz_a4_bc_s + opaque_values.to(tl.float32)
            tl.store(_fz_switch_output_ptr + 6 + tl.arange(0, 2), _fz_a4_tpl_result, mask=(tl.program_id(axis=0) == 0) & (tl.program_id(axis=1) == 0) & (tl.program_id(axis=2) == 0))
        pass


def solve(x: torch.Tensor, weight: torch.Tensor, bias: torch.Tensor, output: torch.Tensor, B: int, L: int, D: int, K: int) -> None:
    _fz_device = x.device
    _fz_switch_ptr = torch.tensor([0, 1], dtype=torch.int32, device=_fz_device)
    _fz_switch_output_ptr = torch.empty(500, dtype=torch.float32, device=_fz_device)
    BLOCK_D = 128
    n_d_blocks = (D + BLOCK_D - 1) // BLOCK_D
    n_l_blocks = B * L
    grid = (n_d_blocks * n_l_blocks,)
    _causal_dw_conv1d_kernel[grid](
        _fz_switch_ptr,
        _fz_switch_output_ptr,
        x,
        weight,
        bias,
        output,
        B,
        L,
        D,
        K,
        x.stride(0),
        x.stride(1),
        x.stride(2),
        weight.stride(0),
        weight.stride(1),
        n_l_blocks,
        BLOCK_D=BLOCK_D,
    )
    torch.npu.synchronize()


def main() -> int:
    assert torch.npu.is_available()
    torch.manual_seed(0)
    B, L, D, K = 5, 4, 2, 3
    x = torch.randn(B, L, D, dtype=torch.float32, device="npu")
    weight = torch.randn(D, K, dtype=torch.float32, device="npu")
    bias = torch.randn(D, dtype=torch.float32, device="npu")
    bad_runs = 0
    for index in range(10):
        output = torch.full((B, L, D), float("nan"), dtype=torch.float32, device="npu")
        solve(x, weight, bias, output, B, L, D, K)
        unwritten = int(torch.isnan(output).sum().item())
        print(f"run {index}: unwritten={unwritten}/{output.numel()}")
        if unwritten:
            bad_runs += 1
    print(f"summary: bad_runs={bad_runs}/10")
    if bad_runs:
        print("[reproduced] the variant drops masked stores (output left unwritten)")
        return 0
    print("[not reproduced] every element was written")
    return 1


if __name__ == "__main__":
    raise SystemExit(main())
```
